src/tests/golden/quickfix_test.go

```diff
@@ -0,0 +1,126 @@
+package golden
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/VKCOM/noverify/src/linttest"
+)
+
+func TestQuickFix(t *testing.T) {
+	s := NewQuickFixTest(t, `testdata/quickfix`)
+	s.runQuickFixTest()
+}
+
+const (
+	expectedExtension = ".fix.expected"
+	fixExtension      = ".fix"
+)
+
+type QuickFixTest struct {
+	t      *testing.T
+	folder string
+}
+
+func NewQuickFixTest(t *testing.T, folder string) QuickFixTest {
+	return QuickFixTest{
+		t:      t,
+		folder: folder,
+	}
+}
+
+func (t *QuickFixTest) runQuickFixTest() {
+	files, err := linttest.FindPHPFiles(t.folder)
+	if err != nil {
+		t.t.Errorf("Error while searching for files in the %s folder: %s", t.folder, err)
+	}
+
+	for _, file := range files {
+		t.t.Run(strings.TrimSuffix(filepath.Base(file), ".php"), func(t *testing.T) {
+			testFileName := file
+			expectedFileName := file + expectedExtension
+			fixedFileName := file + fixExtension
+
+			testFile, err := os.Open(testFileName)
+			if err != nil {
+				t.Errorf("File %s not open: %s", testFileName, err)
+				return
+			}
+			testFileContent, err := ioutil.ReadAll(testFile)
+			if err != nil {
+				t.Errorf("Reading file %s failed: %s", testFileName, err)
+			}
+			testFile.Close()
+
+			expectedFile, err := os.Open(expectedFileName)
+			expectedFileFound := true
+			if err != nil || expectedFile == nil {
+				expectedFileFound = false
+				expectedFile, err = os.Create(expectedFileName)
+				if err != nil {
+					t.Errorf("File creation %s failed: %s", expectedFileName, err)
+				}
+			}
+
+			var expectedFileContent []byte
+			if expectedFileFound {
+				expectedFileContent, err = ioutil.ReadAll(expectedFile)
+				if err != nil {
+					t.Errorf("Reading file %s failed: %s", expectedFileName, err)
+				}
+				expectedFile.Close()
+			}
+
+			fixedFile, err := os.Open(fixedFileName)
```

Comment:
Doing open + close without defer in a big function begs for the resource leak.
I would recommend moving file writing into a separate function, so you can open+defer close+write.
This way, you're guaranteed (more or less) not to leak a file handle.
Also, consider using `iotuil.WriteFile` where possible.
If you don't have to open the file manually, don't do it, so you don't have to close it.